### PR TITLE
[Merged by Bors] - proposals: add tick height from proposals to blocks

### PR DESCRIPTION
**Pull request by dshulyak**
- add highest known base tick height from activation that are referenced in the proposals to the block. this will be used later in the tortoise to prevent adversarial votes using ballot with lower tick heights
- when choosing valid block from multiple valid blocks use the lowest tick height. the idea is that the lower the tick height - the more time honest voters had to vote

### CODE REVIEW

File: common/types/block.go [hunk 1]
```diff
@@ -156,9 +158,14 @@ func ToBlockIDs(blocks []*Block) []BlockID {
 	return ids
 }
 
-// SortBlocks sort blocks by their IDs.
+// SortBlocks sort blocks tick height, if height is equal by lexicographic order.
 func SortBlocks(blks []*Block) []*Block {
-	sort.Slice(blks, func(i, j int) bool { return blks[i].ID().Compare(blks[j].ID()) })
+	sort.Slice(blks, func(i, j int) bool {
+		if blks[i].TickHeight < blks[j].TickHeight {
```
Comment (countvonzero) [inline common/types/block.go]: i think it's worth adding a test here.

**Review by countvonzero — APPROVED** [review]

File: common/types/block.go (hunk 1 above)
Comment (dshulyak) [reply common/types/block.go]: i would rather make it private and move to mesh. it doesn't need to be exported and in common package if it used once

Comment (countvonzero) [reply common/types/block.go]: yes. make sense

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**